software/py/blood_graph.py

```diff
@@ -3,68 +3,292 @@
 #
 #   vanilla_core execution visualizer.
 # 
-#   input: vanilla_stall_trace.log
+#   input: vanilla_operation_trace.log
 #   output: bitmap file (blood.bmp)
 #
-#   @author tommy
+#   @author tommy and borna
 #
 #   How to use:
-#   python blood_graph.py {start_time} {end_time} {timestep} {vanilla.log}
+#   python blood_graph.py --cycle {start_cycle@end_cycle}  
+#                         --abstract {optional} --input {vanilla_operation_trace.log}
 #
-#   ex) python blood_graph.py 6000000 15000000 20 vanilla_stall_trace.log
+#   ex) python blood_graph.py --cycle 6000000@15000000  
+#                             --abstract --input vanilla_operation_trace.log
 #
-#   {start_time}  start_time in picosecond
-#   {end_time}    end_time in picosecond
-#   {timestep}    time step in picosecond
+#   {cycle}       start_cycle@end_cycle of execution
+#   {abstract}    used for abstract simplifed bloodgraph
 # 
 
 
 import sys
 import csv
-from PIL import Image
+import argparse
+from PIL import Image, ImageDraw, ImageFont
+from itertools import chain
 
 
-class BloodGraph:
+DEFAULT_START_CYCLE = 0
+DEFAULT_END_CYCLE   = 500000
+DEFAULT_MODE       = "detailed"
+DEFAULT_INPUT_FILE = "vanilla_operation_trace.log"
```

Comment:
I think you can make these class variables of the argument parser

Reply:
good point, thanks, done!